app/controllers/designs_controller.rb

```diff
@@ -1,2 +1,40 @@
 class DesignsController < ApplicationController
+  before_action :authenticate_user!
+
+  def show
+    @design = Design.find(params[:id])
+  end
+
+  def new
+    @design = Design.new
+  end
+
+  def create
+    @design = Design.new(design_params)
+    if @design.save
+      redirect_to design_path(@design)
+    else
+      render :new
+    end
+  end
```

Comment:
You need a reference to the user, tag and category your are affecting to this design : 
exemple with the dose and cocktail : 
 def create
    @cocktail = Cocktail.find(params[:cocktail_id])
    @dose = Dose.new(safe_params)
    @dose.cocktail_id = params[:cocktail_id]
    @dose.ingredient_id = params[:dose][:ingredient_id]
    if @dose.save
      redirect_to cocktail_path(@dose.cocktail_id)
    else
      render 'doses/new'
    end
  end